File: app_sample_java/src/main/java/com/joelzhu/lib/scanner/java/commands/CommandSit.java
package com.joelzhu.lib.scanner.java.commands;

import com.joelzhu.lib.scanner.annotation.CompileScan;
import com.joelzhu.lib.scanner.java.ICommand;
import com.joelzhu.lib.scanner.java.utils.Constants;

/**
 * Command: human to sit.
 *
 * @author JoelZhu
 * @since 2023-01-27
 */
@CompileScan(group = Constants.GROUP_REST, priority = 2)
public final class CommandSit implements ICommand {
    @Override
    public String execute() {
        return "Sitting... By default, priority = 2";
    }
}

File: app_sample_kotlin/src/main/java/com/joelzhu/lib/scanner/kotlin/commands/CommandSit.kt
package com.joelzhu.lib.scanner.kotlin.commands

import com.joelzhu.lib.scanner.annotation.CompileScan
import com.joelzhu.lib.scanner.kotlin.ICommand

/**
 * Command: human to sit.
 *
 * @author JoelZhu
 * @since 2023-03-13
 */
@CompileScan(priority = 2)
class CommandSit : ICommand {
    override fun execute(): String {
        return "Sitting... Priority = 2"
    }
}